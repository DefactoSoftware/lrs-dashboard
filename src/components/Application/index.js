import React from 'react';
import CSS from './style.css';

export default ({ children })=>(
  <div className={CSS.App}>
    <div className={CSS.header}>
      <h2 className={CSS.title}>Learning Record Store</h2>
    </div>
    <div className={CSS.content}>
      {children}
    </div>
  </div>
);
