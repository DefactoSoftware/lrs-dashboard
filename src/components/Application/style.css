.App {
  text-align: center;
}

.title {
  margin: 20px 0;
  font-size: 2rem;
}

.header {
  background-color: #333;
  height: 150px;
  padding: 20px;
  color: white;
}

.intro {
  font-size: large;
}
