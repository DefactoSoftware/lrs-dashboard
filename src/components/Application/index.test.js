import React from 'react';
import Application from './index';
import renderer from 'react-test-renderer';

jest.mock('react-router', ()=> ({ Link: ({ children, to })=> (<a href={to}>{children}</a>) }));

it('matches the render snapshot', () => {
  const tree = renderer.create(<Application />).toJSON();

  expect(tree).toMatchSnapshot();
});
