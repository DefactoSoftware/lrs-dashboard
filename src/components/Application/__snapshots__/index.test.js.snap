exports[`test matches the render snapshot 1`] = `
<div
  className="Application">
  <div
    className="Header">
    <h2
      className="title">
      Learning Record Store
    </h2>
    <nav>
      <ul
        className="NavigationList">
        <li
          className="NavigationItem">
          <a
            href="/users">
            Users
          </a>
        </li>
        <li
          className="NavigationItem">
          <a
            href="/statements/graph">
            Statements Graph
          </a>
        </li>
        <li
          className="NavigationItem">
          <a
            href="/statements/list">
            Statements List
          </a>
        </li>
      </ul>
    </nav>
  </div>
  <div
    className="content" />
</div>
`;
